pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MultiStageOperator.java

```diff
@@ -167,6 +173,26 @@ protected TransferableBlock updateEosBlock(TransferableBlock upstreamEos, StatMa
     return upstreamEos;
   }
 
+  @Override
+  public ExplainInfo getExplainInfo() {
+    return new ExplainInfo(getExplainName(), getExplainAttributes(), getChildrenExplainInfo());
+  }
+
+  protected List<ExplainInfo> getChildrenExplainInfo() {
+    return getChildOperators().stream()
+        .filter(Objects::nonNull)
+        .map(Operator::getExplainInfo)
+        .collect(Collectors.toList());
+  }
+
+  protected String getExplainName() {
+    return toExplainString();
+  }
+
+  protected Map<String, Plan.ExplainNode.AttributeValue> getExplainAttributes() {
+    return Collections.emptyMap();
+  }
```

Comment:
That is what I tried on the first iteration but... these method should be public, which is something I don't like because that is not part of the `Operator` interface. We could create another abstract class that extended by both `BaseOperator` and `MultiStageOperator`. I'm open to add that change, but I think this is simpler for now.